extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultTransitManager.java

```diff
@@ -376,20 +383,55 @@ public VaultTransitKeyDetail readKey(String keyName) {
     }
 
     protected VaultTransitKeyDetail map(VaultTransitReadKeyData data) {
-        VaultTransitKeyDetail result = new VaultTransitKeyDetail();
+        VaultTransitKeyVersionData latestVersionData = data.keys.get(Integer.toString(data.latestVersion));
+        VaultTransitKeyDetail result;
+        if (latestVersionData.publicKey != null) {
+            Map<String, VaultTransitAsymmetricKeyVersion> versions = data.keys.entrySet().stream()
+                    .collect(Collectors.toMap(
+                            Entry::getKey,
+                            entry -> {
+                                VaultTransitKeyVersionData keyVersionData = entry.getValue();
```

Comment:
please create dedicated methods such as `createAsymmetricKeyVersio(VaultTransitKeyVersionData)` so that we can have one liners lambdas.
as discussed elsewhere, I am not convinced by the `VaultTransitSymmetricKeyDetail` and `VaultTransitAsymmetricKeyDetail` classes.